app/controllers/users_controller.rb

```diff
@@ -130,22 +132,34 @@ def full_delete
     end
   end
 
-  def remove_association
-    provider = params[:provider]
-    identity = @user.identities.find_by(provider: provider)
+  def remove_identity
     set_tabs("account")
 
-    if @user.identities.size == 2 && identity
+    error_message = "An error occurred. Please try again or send an email to: #{SiteConfig.email_addresses[:default]}"
+    unless Authentication::Providers.enabled?(params[:provider])
+      flash[:error] = error_message
+      redirect_to user_settings_path(@tab)
+      return
+    end
+
+    provider = Authentication::Providers.get!(params[:provider])
```

Comment:
could use your new helper here `authentication_provider(provider_name)`

Reply:
It's a view helper, so you'd need to call it as`helpers.authentication_provider(params[:provider])`.